sympy/matrices/expressions/fourier.py

```diff
@@ -21,6 +23,32 @@ def _entry(self, i, j, **kwargs):
     def _eval_inverse(self):
         return IDFT(self.n)
 
+    def _eval_determinant(self):
+        eigs = self._eval_eigenvals()
+        if eigs:
+            tmp = 1
+            for e, cnt in eigs.items():
+                tmp *= e**cnt
+            return tmp
+        return Determinant(self)
+
+    def _eval_eigenvals(self):
+        # Based on Dickinson and Steiglitz, "Eigenvectors and Functions of the
+        # Discrete Fourier Transform", 1982
+        if self.n.is_Integer:
+            m = self.n // 4
+            r = self.n % 4
+            if r == 0:
+                ret = {1: m+1, -1: m, I: m-1, -I: m}
```

Comment:
I wonder if these unsympified ints can leak out

Reply:
I originally planned to sympify them, but then I wasn't sure what the current tradition was. Both keys and vals?

Reply:
The values should be plain `int` I think.